sdk/tests/platform/test_az_mqtt_policy.c

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <az_test_precondition.h>
+#include <azure/core/az_event.h>
+#include <azure/core/az_event_policy.h>
+#include <azure/core/az_mqtt.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_config_internal.h>
+#include <azure/core/internal/az_event_pipeline_internal.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <azure/core/internal/az_mqtt_policy.h>
+#include <azure/core/internal/az_precondition_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+void az_platform_critical_error(void) { assert_true(false); }
+
+// MQTT information for testing
+#define TEST_MQTT_ENDPOINT "test.mosquitto.org"
+#define TEST_MQTT_PORT 1884 // Unencrypted, authenticated port.
+#define TEST_MQTT_USERNAME "rw"
+#define TEST_MQTT_PASSWORD "readwrite"
+#define TEST_MQTT_CLIENT_ID "mqtt_client_id_0505"
```

Comment:
Why this specific client id?

Reply:
This is indeed a problem: we won't be able to run these tests in parallel (the broker will disconnect the older client with the same client ID). We will solve this issue by deploying a local MQTT broker to test against.